lib/devise/models/confirmable.rb

```diff
@@ -169,6 +169,7 @@ def skip_reconfirmation!
         # in models to map to a nice sign up e-mail.
         def send_on_create_confirmation_instructions
           send_confirmation_instructions
+          skip_reconfirmation!
```

Comment:
@printercu Hi, it is used in the case that a recored is created with #save called in callbacks. It prevents the email to be sent twice in that case.
However, this method is only called in `after_create` so it doesn't affect your update.